test/src/test/java/hudson/slaves/SlaveComputerTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, suren <zxjlwt@126.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.slaves;
+
+import com.gargoylesoftware.htmlunit.WebResponse;
+import hudson.model.*;
+import hudson.security.ACL;
+import hudson.security.ACLContext;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONNull;
+import net.sf.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.MockAuthorizationStrategy;
+import org.xml.sax.SAXException;
+
+import java.io.IOException;
+
+/**
+ * @author suren
+ */
+public class SlaveComputerTest {
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void testGetAbsoluteRemotePath() throws Exception {
+        Node nodeA = j.createOnlineSlave();
+        String path = ((DumbSlave) nodeA).getComputer().getAbsoluteRemotePath();
+        Assert.assertNotNull(path);
+        Assert.assertEquals(getRemoteFS(nodeA, null), path);
+
+        String userAlice = "alice";
+        MockAuthorizationStrategy authStrategy = new MockAuthorizationStrategy();
+        authStrategy.grant(Computer.CONFIGURE, Jenkins.READ).everywhere().to(userAlice);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(authStrategy);
+        try(ACLContext context = ACL.as(User.getById(userAlice, true))) {
+            nodeA = j.createOnlineSlave();
+            path = ((DumbSlave) nodeA).getComputer().getAbsoluteRemotePath();
+            Assert.assertNull(path);
+            Assert.assertNull(getRemoteFS(nodeA, userAlice));
```

Comment:
So the point here is that, because alice does not have Computer.CONNECT, she does not see the path?

Would be nice to extend this to bob, who does have Computer.CONNECT, and can see the path. It's sort of part of the first part of this test, but since that has no security at all, there's no assertion for the correct permission check.